src/Event/Subscriber/BddSubscriber.php

```diff
@@ -203,10 +203,22 @@ public function afterStep(AfterStepTested $event)
             }
         }
 
-        $imageFileName = $this->takeScreenshot();
-        $httpTransactions = $this->getHttpTransactionEvents();
+        if (($resultCode == TestResult::FAILED) AND (Athena::settings()->get('bddScreenshotOnlyOnFailed')->orDefaultTo(true) == true)){
```

Comment:
Instead of calling `Athena::settings()->get('bddScreenshotOnlyOnFailed')->orDefaultTo(true)` all the times, it's best to store the result of the method call in a variable.